### PR TITLE
sql: more assorted telemetry fixes

### LINKED ISSUE #31990: sql: add github issue number to the following unimplemented features
**Describe the problem**

Add github issue tracking number to the following unimplemented features:

- [x] unimplemented.alter set type | #9851 with details
- [x] unimplemented.alter set non null | #28751
- [x] unimplemented.alter using |  #9851 with details
- [x] unimplemented.create sequence OWNED BY option | #26382
- [x] unimplemented.at tz |  #32005
- [x] unimplemented.on conflict on constraint | #28161
- [x] unimplemented.match simple | #20305 
- [x] unimplemented.frame_bound |  Implemented in v2.1 via window function improvements
- [x] unimplemented.func_application |  Implemented in v2.1 via window function improvements
- [x] unimplemented.alter add fk | #8855
- [x] unimplemented.with recursive | #21085
- [x] unimplemented.alter add computed | Implemented in v2.1
- [x] unimplemented.set local | #32562
- [x] unimplemented.index_elem a_expr | #9682
- [x] unimplemented.func_expr_common_subexpr | #32563 
- [x] unimplemented.func const | #32556 
- [x] unimplemented.create sequence AS option | #25110 with details
- [x] unimplemented.alter add check | #29639
- [x] unimplemented.alter column type | #9851 with details
- [x] unimplemented.pg_catalog.pg_matviews | N/A
- [x] unimplemented.pg_catalog.pg_statio_user_tables | N/A
- [x] unimplemented.import.mysqlcoltype.GEOMETRY |  #32560, #32559 and #32561

### PR DESCRIPTION
**Pull request by knz**
Fixes #31990. Requested by @awoods187.

### CODE REVIEW

**Review by BramGruneir — APPROVED** [review]
:lgtm:

with some minor comments

> Reviewed 2 of 2 files at r1, 2 of 2 files at r2, 3 of 3 files at r3, 2 of 2 files at r4, 8 of 8 files at r5, 2 of 2 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 2 of 2 files at r10, 1 of 1 files at r11, 2 of 2 files at r12, 2 of 2 files at r13, 3 of 3 files at r14, 3 of 3 files at r15, 1 of 1 files at r16.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-:-LSBreZJ7ULeRCnLFhGn:bt86nmq)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/ccl/importccl/read_import_mysql.go, line 614 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-LSBdFqsAr8uBHO-8ZDs:-LSBdFqt1vocGkMZgoXg:bs8vitx) ([raw file](https://github.com/cockroachdb/cockroach/blob/7d0f348655c093f4d94709df8b1973815bd43bc0/pkg/ccl/importccl/read_import_mysql.go#L614)):*
> ```Go
> 		def.Type = coltypes.JSON
> 
> 	case mysqltypes.Set:
> ```

Is there a test for these?
___
*[pkg/sql/sequence.go, line 245 at r16](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-LSBrZhU5RzccoHLY-vF:-LSBrZhU5RzccoHLY-vG:b4u25rr) ([raw file](https://github.com/cockroachdb/cockroach/blob/c1fd2dcaf625ac62354d33937cd97f6c33cdacaa/pkg/sql/sequence.go#L245)):*
> ```Go
> 				// Do nothing; this is the default.
> 			case v > 1:
> 				return pgerror.UnimplementedWithIssueErrorf(32567,
> ```

tests for these?
___
*[pkg/sql/parser/sql.y, line 6812 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-LSBrQQr6HVZTgK_GrlB:-LSBrQQr6HVZTgK_GrlC:b684qbi) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5aa4ea9497ea019589f91edec2de7e0b6aec9d3/pkg/sql/parser/sql.y#L6812)):*
> ```Yacc
> 
> opt_timezone:
>   WITH_LA TIME ZONE	{ $$.val = true; }
> ```

leftover tabs

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -6764,38 +6785,34 @@ const_datetime:
...
-| TIMESTAMP WITH_LA TIME ZONE
-  {
-    $$.val = coltypes.TimestampWithTZ
-  }
+| TIMESTAMPTZ '(' ICONST ')'            { return unimplementedWithIssue(sqllex, 32098) }
+
+opt_timezone:
+  WITH_LA TIME ZONE	{ $$.val = true; }
+| WITHOUT TIME ZONE	{ $$.val = false; }
+| /*EMPTY*/			{ $$.val = false; }
```
Comment (mjibson) [inline pkg/sql/parser/sql.y]: nit: whitespaces appear incorrect here

**Review by madelynnblue — APPROVED** [review]

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32556)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/ccl/importccl/read_import_mysql.go, line 614 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-LSBdFqsAr8uBHO-8ZDs:-LSFTBCqAlMZxzaMBRXa:bz4ww2a) ([raw file](https://github.com/cockroachdb/cockroach/blob/7d0f348655c093f4d94709df8b1973815bd43bc0/pkg/ccl/importccl/read_import_mysql.go#L614)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Is there a test for these?
</blockquote></details>

Filed #32592 to follow up.
___
*[pkg/sql/sequence.go, line 245 at r16](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-LSBrZhU5RzccoHLY-vF:-LSFR_2q4dAitsij71iy:bpfua9w) ([raw file](https://github.com/cockroachdb/cockroach/blob/c1fd2dcaf625ac62354d33937cd97f6c33cdacaa/pkg/sql/sequence.go#L245)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

tests for these?
</blockquote></details>

Yes they were present previously already in `logic_test/sequences`: see `statement error pgcode 0A000 ...`
___
*[pkg/sql/parser/sql.y, line 6812 at r15](https://reviewable.io/reviews/cockroachdb/cockroach/32556#-LSBrQQr6HVZTgK_GrlB:-LSFRsHlE2NzOPvZ8SmR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5aa4ea9497ea019589f91edec2de7e0b6aec9d3/pkg/sql/parser/sql.y#L6812)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

leftover tabs
</blockquote></details>

Done.

**Comment by knz** [comment]
TFYRs!

bors r+

**Pull request merged**